rtmr/extend/extend.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package extend
+
+import (
+	"crypto"
+	_ "crypto/sha512" // Registrer SHA384 and SHA512
+	"fmt"
+	"github.com/google/go-configfs-tsm/configfs/configfsi"
+	"github.com/google/go-configfs-tsm/configfs/linuxtsm"
+	"github.com/google/go-configfs-tsm/rtmr"
+)
+
+// ExtendDigestClient extends the measurement to the rtmr with the given digest.
+func ExtendDigestClient(client configfsi.Client, rtmrIndex int, digest []byte) error {
```

Comment:
I think we should also include these public APIs in the Readme as well.

Reply:
Yes, I will submit a new PR to add these APIs in the Readme.